controllers/account.go

```diff
@@ -115,7 +118,29 @@ func (c *ApiController) Signup() {
 	userId := fmt.Sprintf("%s/%s", form.Organization, form.Username)
 	msg := object.CheckUserSignup(form.Organization, form.Username, form.Password, form.Name, form.Email, form.Phone, form.Affiliation)
 	if msg != "" {
-		resp = Response{Status: "error", Msg: msg, Data: ""}
+		if form.WithProvider {
+			user := &object.User{
+				Owner:       form.Organization,
+				Name:        form.Username,
+				DisplayName: form.Name,
+				Avatar:      form.Avatar,
+				Password:    form.Password,
+				Email:       form.Email,
+				Phone:       form.Phone,
+				Affiliation: form.Affiliation,
+			}
+			user.UpdateUserHash()
+			user.PreHash = user.Hash
+
+			object.UpdateUserForOriginal(user)
+
+			object.DisableVerificationCode(form.Email)
+			object.DisableVerificationCode(checkPhone)
+			util.LogInfo(c.Ctx, "API: [%s] is signed up as new user", userId)
+			resp = Response{Status: "ok", Msg: "", Data: userId}
+		} else {
+			resp = Response{Status: "error", Msg: msg, Data: ""}
+		}
```

Comment:
These lines do not add the new user to the database. `object.UpdateUserForOriginal` can only update user, but not insert a new user.

And I think we can not add a user here, because when `msg != ""`, the new user information is invalid, and we should not add an invalid user to the database.

Reply:
At f672045, already adds user basic info(like org, owner, id) at `controllers/auth.go` line `264` if the application enable user signup
These lines only for complete and validate user info from OAuth if they want.